plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -606,6 +608,25 @@ public Iterator<TableColumnsMetadata> streamTableColumns(ConnectorSession sessio
                 .iterator();
     }
 
+    private static Schema getCurrentSnapshotSchema(Table icebergTable)
+    {
+        Optional<Long> tableSnapshotId = Optional.ofNullable(icebergTable.currentSnapshot()).map(Snapshot::snapshotId);
+        if (tableSnapshotId.isPresent() && !findLatestSnapshotId(icebergTable.snapshots()).equals(tableSnapshotId)) {
```

Comment:
why `!findLatestSnapshotId` check is necessary?
please document

is it test-covered?

Reply:
For instance, when changed the table definition (e.g. `ADD COLUMN`), `tableSnapshotId` still points to the old definition before adding the column. The condition is for using the latest schema in such cases. 

Added a code comment and it's covered in `ALTER TABLE` → `NSERT INTO` in `testRollbackSnapshotWithSchemaEvolution`.

Reply:
I get it now. However, consider the case

```sql
CREATE TABLE t (a varchar);
INSERT INTO t  VALUES 'one column' ; -- snapshot "s"

-- returns `a` column
SELECT * FROM t FOR VERSION AS OF s; 

ALTER TABLE t ADD COLUMN (b varchar);

--  this should return a, b columns:
SELECT * FROM t;

-- and this should continue to return only the `a` column
SELECT * FROM t FOR VERSION AS OF s; 
```

Can you verify this is the case?

Reply:
I think #14076 covered the case. Let me know if my understanding is correct. Confirmed it works as
```sql
CREATE TABLE t (a varchar);
INSERT INTO t  VALUES 'one column' ;

TABLE "t$snapshots";
            committed_at            |     snapshot_id     |      parent_id      | operation |                                                >
------------------------------------+---------------------+---------------------+-----------+------------------------------------------------>
 2022-09-13 15:24:59.540 Asia/Tokyo | 5659256701260247696 |                NULL | append    | /var/folders/8s/dkvf18z55lj_9yxhy1n54sph0000gn/>
 2022-09-13 15:24:59.744 Asia/Tokyo |  261758948915485069 | 5659256701260247696 | append    | /var/folders/8s/dkvf18z55lj_9yxhy1n54sph0000gn/>

-- returns `a` column
SELECT * FROM t FOR VERSION AS OF 261758948915485069;
     a
------------
 one column

ALTER TABLE t ADD COLUMN b varchar;
ADD COLUMN

--  this should return a, b columns:
SELECT * FROM t;
     a      |  b
------------+------
 one column | NULL

-- and this should continue to return only the `a` column
SELECT * FROM t FOR VERSION AS OF 261758948915485069;
     a
------------
 one column
```